pkgs/build-support/fetchurl/default.nix

```diff
@@ -124,6 +124,8 @@ if (!hasHash) then throw "Specify hash for fetchurl fixed-output derivation: ${s
 
   inherit curlOpts showURLs mirrorsFile impureEnvVars postFetch downloadToTemp executable;
 
+  userAgent = "Nix ${builtins.nixVersion} on nixpkgs ${stdenv.lib.nixpkgsVersion}";
```

Comment:
Doing this will cause all `.drv` files to change every time you update Nixpkgs, which is a lot of unnecessary I/O.

Also, shouldn't this be the other way around ("Nixpkgs on Nix")?